### PR TITLE
add nft card component

### LINKED ISSUE #1678: Create a "NFT card" component
### Description

Based on information gathered from @KubaMikolajczyk and @TCzechowski, as part of the NFT MVP scope of work:
- https://www.figma.com/file/uY98FOmQPgozpNOtAiklBg/Issuing-NFT?node-id=1063%3A67422
- https://www.figma.com/file/yVS7nkKjUtCeBcjUDqBbCF/Transaction?node-id=660%3A33264

### Figma link

_No response_

### How urgent this is?

This will block me soon

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #1678

### CODE REVIEW

File: src/components/NftCard/NftCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      { assetUrl: 'https://placedog.net/100/100?random=5', tooltipText: 'Jeff' },
+      { assetUrl: 'https://placedog.net/100/100?random=6', tooltipText: 'Alan' },
+      { assetUrl: 'https://placedog.net/100/100?random=7', tooltipText: 'Marietta' },
+    ],
+    owner: { assetUrl: 'https://placedog.net/100/100?random=8', tooltipText: 'John' },
+  },
+} as Meta
+
+const Template: Story<NftCardProps> = (args) => (
+  <div style={{ width: '360px' }}>
```
Comment (drillprop) [inline src/components/NftCard/NftCard.stories.tsx]: shouldn't we apply this width to NFTCard itself?

File: src/components/NftCard/Members.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+type MembersProps =
+  | {
+      caption: 'Creator' | 'Owner'
+      members: Member
+    }
+  | { caption: 'Supporters'; members: Member[] }
+
+export const Members: React.FC<MembersProps> = ({ caption, members }) => {
+  const avatars =
+    members && Array.isArray(members)
```
Comment (drillprop) [inline src/components/NftCard/Members.tsx:30]: we used `Array.isArray(members)` two times. Let's store this boolean in a variable maybe?

File: src/components/NftCard/NftCard.styles.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      text-align: right;
+      justify-self: flex-end;
+    }
+  }
+`
+
+export const Separator = styled.hr`
+  grid-column: 1 / span 2;
+  background-color: ${cVar('colorBorderMutedAlpha')};
+  border: 0;
```
Comment (drillprop) [inline src/components/NftCard/NftCard.styles.ts:65]: Why? Does the hr tag have a default border?

File: src/components/NftCard/NftCard.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <NftCard {...args} />
+  </div>
+)
+
+export const Default = Template.bind({})
```
Comment (drillprop) [inline src/components/NftCard/NftCard.stories.tsx:41]: Could you also add a template with no supporters as we have in Figma?
image

**Review by drillprop — CHANGES_REQUESTED** [review]

File: src/components/NftCard/NftCard.stories.tsx (hunk 1 above)
Comment (rafalpawlow) [reply src/components/NftCard/NftCard.stories.tsx]: @toiletgranny does Nft Card should have fixed width?

Comment (kdembler) [reply src/components/NftCard/NftCard.stories.tsx]: Screenshot 2022-01-10 at 15 04 16

Seems so

Comment (toiletgranny) [reply src/components/NftCard/NftCard.stories.tsx]: @rafalpawlow @kdembler Exactly! There's nothing inside of the NFT card component that could determine its width. And so the width should be fixed, with an option of being adjusted depending on the use case (eg. on one page we might want to set the width of this component to be of 3 columns, whereas on another page we might want it to span across 2 columns only.)

File: src/components/NftCard/NftCard.styles.ts (hunk 3 above)
Comment (rafalpawlow) [reply src/components/NftCard/NftCard.styles.ts:65]: Yes, `hr` has default border

**Review by drillprop — APPROVED** [review]
~Not sure if we actually need the `fullWidth` variant, I'd use just `max-width` with `360px`(instead of `width`). This needs to be consulted with Adam(maybe I didn't catch his intentions correctly).~
edit: ignore the above ☝🏻 , we resolved this on rocket chat.
Besides that, looks good to me 👍🏻

File: src/components/NftCard/Members.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+
+import {
+  AvatarGroupWrapper,
+  AvatarWrapper,
+  Caption,
+  MembersWrapper,
+  StyledAvatar,
+  StyledAvatarGroup,
+} from '@/components/NftCard/NftCard.styles'
```
Comment (kdembler) [inline src/components/NftCard/Members.tsx]: Should use relative import

**Review by kdembler — COMMENTED** [review]
The component in the Storybook is missing labels for Creator/Owner?

**Comment by rafalpawlow** [comment]
@kdembler my bad, please check again

**Review by kdembler — APPROVED** [review]
Looks good to me but let's get final approval from @toiletgranny

**Review by toiletgranny — APPROVED** [review]

**Pull request merged**